### PR TITLE
Patvars may be backquoted [ci: last-only]

### LINKED ISSUE #11406: Unable to refer to existing type starting with a lower-case letter
In this example there's a `Var` which is potentially expensive to update, so updates are chained together.

```scala
class Var[T](private var value: T) {
  def get: T = value
  def update(newValue: T) = value = newValue
}

type UpdatePair[T] = (Var[T], T => T)
```

This works:
```scala
  def doUpdates(updateChain: List[UpdatePair[_]]) = {
    updateChain.groupBy(_._1) foreach {
      case (v: Var[t], ops) =>
        var current: t = v.get
        // This is what I want to get rid of:
        type Op = t => t
        ops foreach {
          case (_, op: Op) =>
            current = op(current)
        }
        v() = current
    }
  }
```

Removing the `Op` type alias does not work: 
```scala
ops foreach {
  case (_, op: Function1[`t`, `t`]) =>
  current = op(current)
}
```

This also doesn't work:
```scala
ops foreach {
  case (_, op: (`t`) => `t`) =>
  current = op(current)
}
```

Furthermore, the fact that I even need the `t` type alias seems unnecessary!

### PR DESCRIPTION
**Pull request by som-snytt**
Fixes scala/bug#11406

### CODE REVIEW

File: src/compiler/scala/tools/nsc/ast/parser/Parsers.scala [hunk 1]
```diff
@@ -857,9 +857,14 @@ self =>
 
     /** Convert (qual)ident to type identifier. */
     def convertToTypeId(tree: Tree): Tree = atPos(tree.pos) {
-      convertToTypeName(tree) getOrElse {
-        syntaxError(tree.pos, "identifier expected", skipIt = false)
-        errorTypeTree
+      convertToTypeName(tree) match {
+        case Some(t) =>
+          if (tree.hasAttachment[BackquotedIdentifierAttachment.type])
+            t.updateAttachment(BackquotedIdentifierAttachment)
```
Comment (lrytz) [inline src/compiler/scala/tools/nsc/ast/parser/Parsers.scala]: maybe change `convertToTypeName` to use `treeCopy` instead?

**Review by lrytz — COMMENTED** [review]
LGTM otherwise!

**Comment by som-snytt** [comment]
reverted to the first fix. Breakage of java lookups is worth pursuing separately.

**Comment by lrytz** [comment]
True it doesn't make sense to copy type and symbol when changing `Ident(termName)` to `Ident(typeName)`, but we can copy the attachments and position.

**Review by lrytz — APPROVED** [review]

**Comment by lrytz** [comment]
`Symbols$CyclicReference: illegal cyclic reference involving object Predef` on travis, taking a look if I can reproduce that.

**Comment by lrytz** [comment]
Doesn't reproduce. Travis does
  - `sbt setupPublishCore publishLocal`
  - `sbt -Dstarr.version=<above> setupValidateTest compile`
  - `sbt -Dstarr.version=<above> setupValidateTest Test/compile`

Somehow the last step recompiles parts of the standard library (1 file in the failed build), not sure where the invalidation comes from. I didn't get any locally on Java 17. The fact that an incremental build of the standard library fails doesn't surprise me, I'm often getting `illegal cyclic reference` in the standard library when switching branches.

**Pull request merged**